src/Data/CAS/RocksDB.hs

```diff
@@ -301,6 +305,44 @@ tableDelete db k = R.delete
     (encKey db k)
 {-# INLINE tableDelete #-}
 
+-- -------------------------------------------------------------------------- --
+-- Batches
+
+-- | A single update operation for a 'RocksDbTable'
+--
+data RocksDbUpdate
+    = forall k v . RocksDbDelete
+        { _rocksDbUpdateTable :: !(RocksDbTable k v)
+        , _rocksDbUpdateKey :: !k
+        }
+    | forall k v . RocksDbInsert
+        { _rocksDbUpdateTable :: !(RocksDbTable k v)
```

Comment:
Better to factor table out so it isn't repeated, e.g.

```
data RocksDbUpdate k v = RocksDbInsert k v | RocksDbDelete k

updateBatch :: Traversable s => Traversable t => s (RocksDbTable k v, t (RocksDbUpdate k v)) -> IO ()
```

Reply:
This would limit the batch to operate on a single table.

Reply:
No, because the batch would look like (with `s = t = []` here

```
updateBatch :: [(RocksDbTable k v, [RocksDbUpdate k v])]
```

Please read a little more carefully. The table pointer is being duplicated in your version _n_ times for no reason.